src/Kestrel.Core/Internal/Http2/Http2MessageBody.cs

```diff
@@ -16,6 +16,11 @@ protected Http2MessageBody(Http2Stream context)
             _context = context;
         }
 
+        protected override void OnReadStarting()
+        {
+            TryProduceContinue();
```

Comment:
It looks to me like the 100-continue response is being sent every time the body of an "Expect: 100-continue" request is read.

Shouldn't this instead behave like HTTP/1.1 where we do not send the 100-continue response if the request body is already available when the app starts reading it?